platform/darwin/docs/guides/Predicates and Expressions.md

```diff
@@ -49,91 +53,96 @@ The following aggregate operators are supported:
-               <code>MGLPointAnnotation</code> class
-           </li>
-           <li>
-               <code>LineString</code> for polyline features, corresponding to
-               the <code>MGLPolyline</code> class
-           </li>
-           <li>
-               <code>Polygon</code> for polygon features, corresponding to the
-               <code>MGLPolygon</code> class
-           </li>
-       </ul>
-   </td>
-</tr>
 <tr>
    <td><code>point_count</code></td>
+   <td>Number</td>
    <td>The number of point features in a given cluster.</td>
 </tr>
 </tbody>
 </table>
 
-For floating-point values, use `-[NSNumber numberWithDouble:]` instead of 
-`-[NSNumber numberWithFloat:]` to avoid precision issues.
+Some characters may not be used directly as part of a key path in a format
+string. For example, if a feature’s attribute is named `ISO 3166-1:2006`, an
+expression format string of `lowercase(ISO 3166-1:2006)` or a predicate format
+string of `ISO 3166-1:2006 == 'US-OH'` would raise an exception. Instead, use a
+`%K` placeholder or the `+[NSExpression expressionForKeyPath:]` initializer:
 
-Automatic type casting is not performed. Therefore, a feature only matches this
-predicate if its value for the attribute in question is of the same type as the
-value specified in the predicate. Also, operator modifiers such as `c` (for
-case insensitivity), `d` (for diacritic insensitivity), and `l` (for locale
-sensitivity) are unsupported for comparison and aggregate operators that are
-used in the predicate.
-
-It is possible to create expressions that contain special characters in the
-predicate format syntax. This includes the `$` in the `$featureIdentifier` and 
-`$geometryType` special style attributes and also `hyphen-minus` and `tag:subtag`. 
-However, you must use `%@` in the format string to represent these variables:
-`@"$geometryType == 'LineString'"`.
-
-## Using expressions to configure layout and paint attributes
-
-### Key paths
+```objc
+[NSPredicate predicateWithFormat:@"%K == 'US-OH'", @"ISO 3166-1:2006"];
+[NSExpression expressionForFunction:@"lowercase:"
+                          arguments:@[[NSExpression expressionForKeyPath:@"ISO 3166-1:2006"]]]
+```
 
-A key path expression refers to an attribute of the `MGLFeature` object being
-evaluated for display. For example, if a polygon’s `MGLFeature.attributes`
-dictionary contains the `floorCount` key, then the key path `floorCount` refers
-to the value of the `floorCount` attribute when evaluating that particular
-polygon.
+```swift
+NSPredicate(format: "%K == 'US-OH'", "ISO 3166-1:2006")
+NSExpression(forFunction: "lowercase:",
+             arguments: [NSExpression(forKeyPath: "ISO 3166-1:2006")])
+```
 
-### Predefined functions
+### Functions
 
 Of the
 [functions predefined by the `+[NSExpression expressionForFunction:arguments:]` method](https://developer.apple.com/documentation/foundation/nsexpression/1413747-init#discussion),
```

Comment:
I know that what text should be linked is a debated topic, but since this is a link to the docs for that specific method, it struck me as a little odd to see the surrounding prose linked with the code-blocked method name. I’d advocate for just linking the method itself.

Reply:
I linked it this way because the table of predefined functions is below the fold.